plugins/main/public/components/common/wazuh_discover/wz_discover.tsx

```diff
@@ -0,0 +1,247 @@
+  EuiPanel,
+  EuiDataGridColumn
+} from '@elastic/eui';
+import { IntlProvider } from 'react-intl';
+import { IndexPattern } from '../../../../../../src/plugins/data/common';
+import { SearchResponse } from '../../../../../../src/core/server';
+import { useDocViewer } from '../doc_viewer';
+import DocViewer from '../doc_viewer/doc_viewer';
+import { DiscoverNoResults } from '../../overview/vulnerabilities/common/components/no_results';
+import { LoadingSpinner } from '../../overview/vulnerabilities/common/components/loading_spinner';
+import { useDataGrid, tDataGridColumn, exportSearchToCSV } from '../hooks/data_grid';
+import { ErrorHandler, ErrorFactory, HttpError } from '../../../react-services/error-management';
+import { withErrorBoundary } from '../hocs';
+import { HitsCounter } from '../../../kibana-integrations/discover/application/components/hits_counter';
+import { formatNumWithCommas } from '../../../kibana-integrations/discover/application/helpers';
+import useSearchBar from '../hooks/search_bar/use_search_bar';
+import { search } from '../hooks/search_bar';
+import { getPlugins } from '../../../kibana-services';
+import { ViewMode } from '../../../../../../src/plugins/embeddable/public';
+import { getDiscoverPanels } from './config/chart';
+const DashboardByRenderer = getPlugins().dashboard.DashboardContainerByValueRenderer;
+
+/**
+ * ToDo:
+ * - add possibility to customize column render
+ * - add save query feature
+ */
+
+export const MAX_ENTRIES_PER_QUERY = 10000;
+
+type WazuhDiscoverProps = {
+  indexPatternName: string;
+  tableColumns: tDataGridColumn[];
+}
+
+const WazuhDiscover = (props: WazuhDiscoverProps) => {
+  const { indexPatternName, tableColumns: defaultTableColumns } = props
+  const { searchBarProps } = useSearchBar({
+    defaultIndexPatternID: indexPatternName,
+  })
+  const { isLoading, filters, query, indexPatterns } = searchBarProps;
+  const SearchBar = getPlugins().data.ui.SearchBar;
+  const [results, setResults] = useState<SearchResponse>({} as SearchResponse);
+  const [inspectedHit, setInspectedHit] = useState<any>(undefined);
+  const [indexPattern, setIndexPattern] = useState<IndexPattern | undefined>(undefined);
+  const [isSearching, setIsSearching] = useState<boolean>(false);
+  const [isExporting, setIsExporting] = useState<boolean>(false);
+
+  const onClickInspectDoc = useMemo(() => (index: number) => {
+    const rowClicked = results.hits.hits[index];
+    setInspectedHit(rowClicked);
+  }, [results]);
+
+  const DocViewInspectButton = ({ rowIndex }: EuiDataGridCellValueElementProps) => {
+    const inspectHintMsg = 'Inspect document details';
+    return (
+      <EuiToolTip content={inspectHintMsg}>
+        <EuiButtonIcon
+          onClick={() => onClickInspectDoc(rowIndex)}
+          iconType='inspect'
+          aria-label={inspectHintMsg}
+        />
+      </EuiToolTip>
+    );
+  };
+
+  const dataGridProps = useDataGrid({
+    ariaLabelledBy: 'Vulnerabilities Inventory Table',
```

Comment:
The text refers to vulnerabilities but this will be used in other modules.

Reply:
Thanks @chantal-kelm